### PR TITLE
Implementation notes & new example.

### PR DESCRIPTION
**Pull request by nitin42**
Added a new example using HyperApp. If we are good to go then I can upload the assets and utilities files for the example !

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +33,9 @@
   },
   "devDependencies": {
     "babel-preset-es2015": "^6.24.1",
+    "flow-bin": "^0.48.0",
```
Comment (jbucaran) [inline package.json]: @nitin42 Please remove these. 👍

File: src/h.js [hunk 2]
```diff
@@ -1,4 +1,18 @@
-export default function(tag, data) {
+// @flow
```
Comment (jbucaran) [inline src/h.js]: @nitin42 Please remove these. 👍

File: implementation.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+```
+
+More Examples -
+
+**Creating a `svg`**
+
+```javascript
+$ createElementFrom(h('svg', { width: "100", height: "100" }, h('circle', { cx: '50', cy: '50'})))
+
+<svg width="100" height="100">
```
Comment (jbucaran) [inline implementation.md]: What's going on with the formatting here?

Comment (nitin42) [inline implementation.md]: Nested nodes for creating a svg!

Comment (jbucaran) [inline implementation.md]: No, the markdown is wrong. 

screen shot 2017-06-30 at 14 11 20

Also, what is `$`?

One more thing, don't use ```javascript. Use ```js or ```jsx.

Comment (nitin42) [inline implementation.md]: Ok. Let me see!

File: .flowconfig [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[ignore]
```
Comment (jbucaran) [inline .flowconfig]: Please remove this file. This PR is meant to add docs, not flow. 🙏

File: implementation.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+ <circle cx="50 cy="50"></circle>
+</svg>
+```
+
+**Lifecycle hook**
+
+In this example, style of an element is updated after it is created and added to DOM.
+
+```javascript
```
Comment (jbucaran) [inline implementation.md]: \`\`\`jsx

:wink:

File: implementation.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+It takes an element, oldData and current data on that element to update a value.
+
+Example -
+
+Let's say we have a DOM tree and we diff against its previous version. If the resultant tag (element type) is same in both the trees, it means the values must have been updated (example - forms with controlled components). So we update the element data using the current value, old value and the element itself.
+
+**Inner working (part of `App.js`)**
+
+```javascript
```
Comment (jbucaran) [inline implementation.md]: \`\`\`jsx

:wink:

File: package.json [hunk 7]
```diff
@@ -24,6 +24,7 @@
     "test": "jest --coverage --no-cache",
     "build": "rollup -cm -n hyperapp -f umd -i src/index.js -o dist/hyperapp.js",
     "prepublish": "npm run build",
+    "flow": "./node_modules/.bin/flow",
```
Comment (jbucaran) [inline package.json]: Please send another PR adding flow if you like, but let's keep this PR just for doc changes. Thanks! 👍

File: package.json [hunk 8]
```diff
@@ -32,6 +33,9 @@
   },
   "devDependencies": {
     "babel-preset-es2015": "^6.24.1",
+    "flow-bin": "^0.48.0",
+    "flow-remove-types": "^1.2.1",
```
Comment (jbucaran) [inline package.json]: Won't need any of these either.

**Comment by nitin42** [comment]
I wrote the functions and their output as some sort of pseudo code representation. But anyway I've made the changes. I removed the flow annotations in the earlier commits but they didn't go away 🤔

**Comment by jbucaran** [comment]
@nitin42 Looking much better! Please allow me some time to go over this 🙏 

Feel free to push any other commits you wish, e.g., grammar tweaks, etc.

**Comment by jbucaran** [comment]
@nitin42 I like to keep all official examples in https://codepen.io/hyperapp. If you don't mind I can create a pen with your example and add generous note to indicate it is your contribution.

If you are okay with this, can you please share the full source code? 😄👍

**Comment by nitin42** [comment]
@jbucaran can you give sometime (one day or so) to push the code  ? I accidentally deleted the folder 😞 . Btwn I've made two more examples!! 😄

**Comment by jbucaran** [comment]
@nitin42 Sure!

👍

**Comment by nitin42** [comment]
Here you go - https://github.com/nitin42/HyperApp-Examples. It contains two examples for now but I'll be adding more 😄

**Comment by jbucaran** [comment]
@nitin42 This was a great effort Nitin and I appreciate the hard work you put into this! I've decided I like to take care of writing the implementation notes myself, but I will surely help myself with what you've written and generously thank you in my commit message when I am done.

Closing! 💯😉👍
